charts/rsyncd/templates/deployment.yaml

```diff
@@ -37,20 +45,26 @@ spec:
           readinessProbe:
             tcpSocket:
               port: 873
+          {{- with .Values.resources }}
           resources:
-            {{- toYaml .Values.resources | nindent 12 }}
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
           volumeMounts:
-            - name: datadir
-              mountPath: {{ .Values.configuration.datadir }}
-              readOnly: true
             - name: rsyncd-conf
-              mountPath: /etc/rsyncd.conf
-              subPath: rsyncd.conf
-              readOnly: true
-            {{- if .Values.configurationFiles.jenkinsMotd.override }}
+              mountPath: /etc/rsyncd.d/rsyncd.inc
+              readOnly: true # Default for configmaps
+              subPath: rsyncd.inc
             - name: jenkins-motd
-              mountPath: {{ .Values.configuration.motd_path }}
-              subPath: jenkins-motd
+              mountPath: {{ .Values.configuration.motd.path }}
+              subPath: jenkins.motd
+              readOnly: true
+            {{- range .Values.configuration.components }}
+            - name: rsyncd-conf
+              mountPath: /etc/rsyncd.d/{{ .name }}.conf
+              readOnly: true # Default for configmaps
+              subPath: {{ .name }}.conf
+            - name: datadir-{{ .name }}
+              mountPath: {{ .path }}
               readOnly: true
```

Comment:
```suggestion
              readOnly: true # Default for configmaps
```
Like L55 & L64

Reply:
not for this one: the datadir is not a configmap